T4MVCExtensions/T4Extensions.cs

```diff
@@ -334,9 +334,11 @@ public static string JavaScriptReplaceableUrl(this UrlHelper urlHelper, ActionRe
             string action = token.ToString();
 
             var specificActionUrl = (from r in RouteTable.Routes.OfType<Route>()
-                                     where r.DataTokens != null && r.Defaults != null
-                                     && r.DataTokens.Any(dt => string.Compare(dt.Key, "area", StringComparison.InvariantCultureIgnoreCase) == 0
-                                         && string.Compare(dt.Value.ToString(), area, StringComparison.InvariantCultureIgnoreCase) == 0)
+                                     where r.Defaults != null
+                                     && ((string.IsNullOrEmpty(area) 
+                                        && (r.DataTokens == null || !r.DataTokens.Any(dt => string.Compare(dt.Key, "area", StringComparison.InvariantCultureIgnoreCase) == 0)))
+                                        || (r.DataTokens != null && r.DataTokens.Any(dt => string.Compare(dt.Key, "area", StringComparison.InvariantCultureIgnoreCase) == 0 
+                                            && string.Compare(dt.Value.ToString(), area, StringComparison.InvariantCultureIgnoreCase) == 0)))
```

Comment:
Agree completely - I was just trying to keep to the original code style as much as possible. Now that I know you're amenable, I'll go ahead and refactor to make it clearer.

The code is essentially checking if there's an area specified in the `ActionResult` and if not, matching a route without `DataTokens` collection or without an "area" key; if so, matching a route that has a `DataTokens` collection with a matching "area" key.